cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/SelfAwareJoinDecider.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2017-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.management.cluster.bootstrap
+
+import akka.actor.ActorSystem
+import akka.discovery.SimpleServiceDiscovery.ResolvedTarget
+import akka.event.Logging
+
+import scala.concurrent.Await
+import scala.util.Try
+
+/**
+ * Class for further behavior in a [[akka.management.cluster.bootstrap.JoinDecider]]
+ * leveraging self host logic.
+ */
+private[bootstrap] abstract class SelfAwareJoinDecider(system: ActorSystem, settings: ClusterBootstrapSettings)
+    extends JoinDecider {
+
+  protected val log = Logging(system, getClass)
+
+  private def discoveryTimeout = settings.contactPointDiscovery.selfDiscoveryTimeout
+
+  /** Returns the current `selfContactPoints` as a String for logging, e.g. [127.0.0.1:64714]. */
+  protected def contactPointsString(contactPoints: Set[(String, Int)]): String =
+    contactPoints.map(_.productIterator.mkString(":")).mkString("[", ",", "]")
+
+  /**
+   * The value `ClusterBootstrap(system).selfContactPoints` is set prior to HTTP binding,
+   * during [[akka.management.AkkaManagement.start()]], hence we accept blocking on
+   * this initialization.
+   */
+  private[bootstrap] def selfContactPoints: Set[(String, Int)] =
+    Try(Await.result(ClusterBootstrap(system).selfContactPoints, discoveryTimeout))
```

Comment:
I think it is only ever a bug in our code that means this won't be ready so i don't think we should expose a timeout to the user.